BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
++                packet.setFadeOut((int) Objects.requireNonNull(title.times()).fadeOut().toMillis() / 50);
++                break;
++            default: break;
++        }
++
++        return packet;
++    }
++}
+diff --git a/proxy/src/main/java/net/md_5/bungee/BungeeCord.java b/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
+index 2938f2f1..abccf735 100644
+--- a/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
++++ b/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
+@@ -51,6 +51,13 @@ import java.util.logging.Logger;
+ import lombok.Getter;
+ import lombok.Setter;
+ import lombok.Synchronized;
++// Waterfall start - we're going on an adventure!
++import net.kyori.adventure.audience.Audience;
++import net.kyori.adventure.text.Component;
++import net.kyori.adventure.text.serializer.gson.GsonComponentSerializer;
++import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
++import org.checkerframework.checker.nullness.qual.NonNull;
++// Waterfall end
+ import net.md_5.bungee.api.CommandSender;
+ import net.md_5.bungee.api.Favicon;
+ import net.md_5.bungee.api.ProxyServer;
+@@ -169,7 +176,14 @@ public class BungeeCord extends ProxyServer
+             .registerTypeAdapter( ScoreComponent.class, new ScoreComponentSerializer() )
+             .registerTypeAdapter( SelectorComponent.class, new SelectorComponentSerializer() )
+             .registerTypeAdapter( ServerPing.PlayerInfo.class, new PlayerInfoSerializer() )
+-            .registerTypeAdapter( Favicon.class, Favicon.getFaviconTypeAdapter() ).create();
++            // Waterfall start - we're going on an adventure!
++            // we have to register all these type adapters so Gson recognises that they are all the same
++            .registerTypeAdapter( Favicon.class, Favicon.getFaviconTypeAdapter() )
++            .registerTypeAdapter(Component.class, GsonComponentSerializer.gson().serializer().getAdapter(Component.class))
++            .registerTypeAdapter(net.kyori.adventure.text.TextComponent.class, GsonComponentSerializer.gson().serializer().getAdapter(net.kyori.adventure.text.TextComponent.class))
++            .registerTypeAdapter(net.kyori.adventure.text.TranslatableComponent.class, GsonComponentSerializer.gson().serializer().getAdapter(net.kyori.adventure.text.TranslatableComponent.class))
++            .create();
++            // Waterfall end
+     @Getter
+     private ConnectionThrottle connectionThrottle;
+     private final ModuleManager moduleManager = new ModuleManager();
+@@ -312,7 +326,10 @@ public class BungeeCord extends ProxyServer
+             @Override
+             public void run()
+             {
+-                independentThreadStop( getTranslation( "restart" ), false );
++                // Waterfall start - we're going on an adventure!
++//                independentThreadStop( getTranslation( "restart" ), false );
++                independentThreadStop(Component.text(getTranslation("restart")), false);
```

Comment:
To self: use `LegacyComponentSerializer`, or look in to adding a new `translate` method that returns a `Component`.